src/solvers/equation_solver.cpp

```diff
@@ -15,28 +15,38 @@ EquationSolver::EquationSolver(MPI_Comm comm, const LinearSolverParameters& lin_
 {
   // Preconditioner configuration is too varied, maybe a PrecondParams is needed?
   // Maybe a redesign to better support custom preconditioners as well
-  switch (lin_params.lin_solver) {
-    case LinearSolver::CG:
-      iter_lin_solver_ = std::make_unique<mfem::CGSolver>(comm);
-      break;
-    case LinearSolver::GMRES:
-      iter_lin_solver_ = std::make_unique<mfem::GMRESSolver>(comm);
-      break;
-    case LinearSolver::MINRES:
-      iter_lin_solver_ = std::make_unique<mfem::MINRESSolver>(comm);
-      break;
-    default:
-      SLIC_ERROR("Linear solver type not recognized.");
-      exitGracefully(true);
-  }
-  iter_lin_solver_->SetRelTol(lin_params.rel_tol);
-  iter_lin_solver_->SetAbsTol(lin_params.abs_tol);
-  iter_lin_solver_->SetMaxIter(lin_params.max_iter);
-  iter_lin_solver_->SetPrintLevel(lin_params.print_level);
+  
+  if (lin_params.lin_solver == LinearSolver::SuperLU) {
+    lin_solver_ = std::make_unique<mfem::SuperLUSolver>(comm);
+  } else {
```

Comment:
I think since the LinearSolverParams don't get used here the existing design should be reconsidered.  One option would be to have constructors corresponding to each configuration (just iterative, just SuperLU, iterative + newton, superLU + newton), but that doesn't scale too well.  Another option might be to just have the engines? store a reference to an mfem::Solver and define Solver wrappers for each component (e.g. an iterative solver class that takes a LinearSolverParams by ctor arg, and a nonlinear solver class that takes a NonlinearParams by ctor arg).